src/plugins/discover/public/hooks/use_persisted_data_view.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import { useDiscoverServices } from './use_discover_services';
+import { showConfirmPanel } from './show_confirm_panel';
+
+export const usePersistedDataView = (dataView: DataView) => {
+  const services = useDiscoverServices();
+
+  const persistDataView = useCallback(async () => {
+    try {
+      const response = await services.dataViews.createAndSave({
+        id: dataView.id,
+        title: dataView.title,
+        timeFieldName: dataView.timeFieldName,
+      });
```

Comment:
I've just tested to share (confirmed to persist), and then to create an alert. I was prompted again. So after persisting,` dataView.isPersisted()` still returns false. 
so I think we should replace our used data view by the one returned by `services.dataViews.createAndSave`, then it should work

Reply:
I've created the follow up issue, which could also be resolved with this PR https://github.com/elastic/kibana/issues/139543 in case it's no simple to prevent already saving a saved search with ad-hoc data view